File: src/domain.ts
export interface Fixture {
  date: Date;
  venue: string;
  leage: string;
  homeTeam: string;
  awayTeam: string;
}

export interface FixturesUpdate {
  date: Date;
  venue: string;
  fixtures: Fixture[];
  source: string;
}

File: src/domain.py
"""Domain models for fixtures."""

from dataclasses import dataclass
from datetime import datetime


@dataclass
class Fixture:
    """Represents a single fixture."""

    date: datetime
    venue: str
    league: str
    home_team: str
    away_team: str


@dataclass
class FixturesUpdate:
    """Represents a fixtures update."""

    date: datetime
    venue: str
    fixtures: list[Fixture]
    source: str
